CMakeLists.txt

```diff
@@ -49,22 +65,25 @@ if (NOT SWIFT_SUPPORTS_DISABLE_IMPLICIT_STRING_PROCESSING_MODULE_IMPORT)
 endif()
 
 # Determine the module triple.
-# FIXME: This is a hack. It's all a hack. Windows isn't setting
-# CMAKE_Swift_COMPILER_TARGET.
-if(CMAKE_Swift_COMPILER_TARGET)
-  string(REGEX REPLACE "macosx[0-9]+([.][0-9]+)?" "macos" SWIFT_MODULE_TRIPLE
-    ${CMAKE_Swift_COMPILER_TARGET})
-elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Windows")
-  if(CMAKE_SYSTEM_PROCESSOR MATCHES "x86_64|AMD64")
-    set(SWIFT_MODULE_TRIPLE "x86_64-unknown-windows-msvc")
-  elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|ARM64|arm64")
-    set(SWIFT_MODULE_TRIPLE "aarch64-unknown-windows-msvc")
+if("${SWIFT_HOST_MODULE_TRIPLE}" STREQUAL "")
+  # FIXME: This is a hack. It's all a hack. Windows isn't setting
```

Comment:
Not sure I follow this.  This is meant to be a user controlled option, what were you expecting?  Or is this is a bug CMake that we should also try to get fixed?

Reply:
The comment is just what was already there, I don't think this is a CMake bug. For non-Windows it's all being set via build-script.

Reply:
Just to expand on this... The reason we need the module triple at all is to output the module in the subfolder format rather than the flat format. Ideally we'll get that fixed at some point 😅. `CMAKE_Swift_COMPILER_TARGET` is being set in build-script, but I assume that if this was added, it wasn't set for Windows somewhere (in PR testing?). `SWIFT_HOST_MODULE_TRIPLE` is being passed from the Swift build, so this path would only be taken in the standalone. Not ideal, but also not really anything to do with the fetchcontent change itself.